src/axom/quest/IntersectionShaper.hpp

```diff
@@ -1466,6 +1441,454 @@ class IntersectionShaper : public Shaper
+    // those points.
+    SegmentMesh* newm = new SegmentMesh(m->getDimension(), mint::SEGMENT);
+    newm->reserveNodes(pointcount);
+
+    SLIC_INFO(axom::fmt::format(
+      "{:-^80}",
+      axom::fmt::format(
+        " Checking contour with {} points for degenerate segments ",
+        pointcount)));
+
+    enum
+    {
+      R = 1,
+      Z = 0
+    };
+
+    // Add contour points
+    int polyline_size = 0;
+    const double EPS = m_vertexWeldThreshold;
+    const double EPS_SQ = EPS * EPS;
+    Point2D cur_point, prev_point;
+    for(int i = 0; i < pointcount; ++i)
+    {
+      // Get the current point.
+      m->getNode(i, cur_point.data());
+
+      // Check for degenerate segments
+      if(polyline_size > 0)
+      {
+        using axom::utilities::isNearlyEqual;
+
+        // both r are (nearly) equal to 0
+        if(isNearlyEqual(cur_point[R], 0.0, EPS) &&
+           isNearlyEqual(prev_point[R], 0.0, EPS))
+        {
+          continue;
+        }
+
+        // both Z are (nearly) the same
+        if(isNearlyEqual(cur_point[Z], prev_point[Z], EPS))
+        {
+          continue;
+        }
+
+        // both points are (nearly) the same
+        if(squared_distance(cur_point, prev_point) < EPS_SQ)
+        {
+          continue;
+        }
+      }
+
+      // Add the current point to the new mesh.
+      newm->appendNode(cur_point[Z], cur_point[R]);
+      prev_point = cur_point;
+      polyline_size += 1;
+    }
+
+    // Get the ZR coordinates.
+    double* z = newm->getCoordinateArray(mint::X_COORDINATE);
+    double* r = newm->getCoordinateArray(mint::Y_COORDINATE);
+
+    // Check if we need to flip the points order.
+    // discretize() is only valid if x increases as index increases
+    bool flip = false;
+    if(polyline_size > 1)
+    {
+      if(z[0] > z[1])
+      {
+        flip = true;
+        SLIC_INFO("Order of contour points has been reversed!"
+                  << " Discretization algorithm expects Z values of contour "
+                     "points to be increasing");
+      }
+    }
+
+    SLIC_INFO(axom::fmt::format(
+      "{:-^80}",
+      axom::fmt::format(" Discretizing contour with {} points ", polyline_size)));
+
+    // Flip point order
+    if(flip)
+    {
+      int i = polyline_size - 1;
+      int j = 0;
+      while(i > j)
+      {
+        axom::utilities::swap(z[i], z[j]);
+        axom::utilities::swap(r[i], r[j]);
+        i -= 1;
+        j += 1;
+      }
+    }
+
+    // Now make polyline_size - 1 segments.
+    int numNewSegments = polyline_size - 1;
+    newm->reserveCells(numNewSegments);
+    for(int i = 0; i < numNewSegments; ++i)
+    {
+      IndexType seg[2] = {i, i + 1};
+      newm->appendCell(seg, mint::SEGMENT);
+    }
+
+    return newm;
+  }
+
+  /*!
+   * \brief Compute the area of the polygon given by \a pts.
+   * \param pts The list of points that make up the polygon.
+   * \return The polygon area.
+   */
+  double area(const std::vector<Point2D>& pts) const
```

Comment:
Note: It's a bit difficult to see from github, but if this is only used for computing the discretized circle area, we might consider directly constructing a `primal::Polygon` as we add the 2D points, and calling `Polygon::area()`

See: https://github.com/LLNL/axom/blob/df5b4d3ac4c55775dc949b39bab98e0a30f90551/src/axom/primal/geometry/Polygon.hpp#L141-L182